### PR TITLE
Add funders to a publication - API and UI

### PR DESCRIPTION
**Pull request by shaunajisc**
The purpose of this PR is to create a feature which enables users to add funders to their publication. 

This PR contains the additional schema to store the funders, an amendment to the publication schema to return the funders within the `getPublication` endpoint, new create and delete funder endpoints, tests, a component that handles the addition/ deletion of funders on the UI within the 'publish a publication' process and amendments to global state.

There are two new endpoints that are as follows:

- Create a funder 
Path: `POST /publications/{id}/funders`
which takes a body that looks like this: 
```javascript
{
            city: 'Manchester',
            country: 'United Kingdom',
            ror: '',
            name: 'Manchester Health and Wellbeing Service',
            link: 'https://buzzmanchester.co.uk/'
}
```
-  Delete a funder
Path: `DELETE /publications/{id}/funders/{funder}`
Which takes no body 
- No get endpoint because the funders will be returned within the `getPublication` endpoint, and there are no use cases for when we'd need this returned outside of viewing the publication

## Acceptance Criteria:
- Users can add many funders.
- Required fields: name, city, country_name, link, ror (optional)
- This can be populated via a ROR ID or manually.
- Once a publication is live funders cannot be edited
- Only the submitter of a publication can modify funders.

## Tests:
```
 PASS  src/components/funder/__tests__/createFunder.test.ts (10.787 s)
  create a funder
    ✓ User can add a funder to their DRAFT publication (897 ms)
    ✓ User cannot add a funder to their LIVE publication (594 ms)
    ✓ User cannot add a funder to another DRAFT publication (583 ms)
    ✓ User cannot add a funder to another LIVE publication (594 ms)
    ✓ User must send correct information to create a funder (no name) (549 ms)
    ✓ User must send correct information to create a funder (no city) (579 ms)
    ✓ User must send correct information to create a funder (no country) (583 ms)
    ✓ User must send correct information to create a funder (no link) (640 ms)
```

```
 PASS  src/components/funder/__tests__/deleteFunder.test.ts (7.883 s)
  delete a funder
    ✓ User can delete a funder from a DRAFT publication (812 ms)
    ✓ User cannot delete a funder from a DRAFT publication they are not the owner of (578 ms)
    ✓ User cannot delete a funder from a LIVE publication (538 ms)
```

## Screenshots:
Screenshot 2022-05-13 at 14 23 19

Screenshot 2022-05-13 at 14 06 04

Screenshot 2022-05-13 at 14 09 11

### CODE REVIEW

File: api/serverless.yml [hunk 1]
```diff
@@ -295,3 +295,19 @@ functions:
                   path: ${self:custom.versions.v1}/bookmarks
                   method: get
                   cors: true
+    createFunder:
+        handler: src/components/funder/routes.create
+        warmup: true
```
Comment (AshleyRedmanJisc) [inline api/serverless.yml:300]: Don't think this is required anymore

File: api/serverless.yml [hunk 2]
```diff
@@ -295,3 +295,19 @@ functions:
...
+        events:
+            - http:
+                  path: ${self:custom.versions.v1}/publications/{id}/funders
+                  method: post
+                  cors: true
+    deleteFunder:
+        handler: src/components/funder/routes.destroy
+        warmup: true
```
Comment (AshleyRedmanJisc) [inline api/serverless.yml:308]: Don't think this is required anymore

File: ui/src/components/Publication/Creation/Funders.tsx [hunk 3]
```diff
@@ -0,0 +1,365 @@
...
+
+const Funders: React.FC = (): React.ReactElement => {
+    const funders = Stores.usePublicationCreationStore((state) => state.funders);
+    const updateFunders = Stores.usePublicationCreationStore((state) => state.updateFunders);
+
+    const publicationId = Stores.usePublicationCreationStore((state) => state.id);
+    const user = Stores.useAuthStore((state) => state.user);
+
+    const [method, setMethod] = React.useState<'ror' | 'manual'>('ror');
+    const [ror, setRor] = React.useState<string>('');
```
Comment (AshleyRedmanJisc) [inline ui/src/components/Publication/Creation/Funders.tsx:127]: Don't think we need to specify type here

**Review by AshleyRedmanJisc — CHANGES_REQUESTED** [review]
Three small straight forward comments, but everything looks great.

**Review by AshleyRedmanJisc — DISMISSED** [review]

File: ui/src/components/Publication/Creation/Funders.tsx [hunk 4]
```diff
@@ -0,0 +1,365 @@
...
+            setSubmitLoading(false);
+            updateFunders([...funders, createdFunder]);
+            setName('');
+            setCountry('');
+            setCity('');
+            setLink('');
+            setRor('');
+        } catch (err) {
+            setSubmitLoading(false);
+            console.log(err);
```
Comment (shaunajisc) [inline ui/src/components/Publication/Creation/Funders.tsx:187]: Need to delete this

**Review by nathan-at-jisc — APPROVED** [review]
LGTM

**Pull request merged**